packages/core/src/scripts/migrate-legacy-project.ts

```diff
@@ -0,0 +1,243 @@
+import path from 'path';
+import { promises as fs } from 'fs';
+import type { ManifestData, ImplDeployment } from '../manifest';
+import type { StorageItem, StorageLayout, TypeItem } from '../storage';
+
+const OPEN_ZEPPELIN_FOLDER = '.openzeppelin';
+const MIGRATION_FILE_LOCATION = 'openzeppelin-cli-migration.json';
+
+export async function migrateLegacyProject(): Promise<void> {
+  const manifestList = await findManifests();
+  const { migrations, migrationData } = await migrateManifestFiles(manifestList);
+
+  for (const manifestFile in migrations) {
+    const newManifest = migrations[manifestFile];
+    await writeJSONFile(getNewManifestLocation(manifestFile), newManifest);
+  }
+
+  await writeJSONFile(MIGRATION_FILE_LOCATION, migrationData);
+}
+
+async function migrateManifestFiles(manifestList: string[]): Promise<MigrationOutput> {
+  const migrationData: Record<string, unknown> = {};
+  const migrations: Record<string, ManifestData> = {};
+
+  for (const manifestFile of manifestList) {
+    const oldManifest = JSON.parse(await fs.readFile(manifestFile, 'utf8'));
+    const { manifestVersion } = oldManifest;
+
+    if (manifestVersion === undefined) {
+      continue;
+    }
+
+    const [major] = manifestVersion.split('.');
+
+    if (major !== '2') {
+      continue;
```

Comment:
Same here, we should validate that the manifest we're migrating is the latest version, otherwise we should fail, the user should use the CLI to obtain an updated manifest, and then we can work with that.

Reply:
I thought of this behavior to skip already migrated files in the same spirit of idempotence of the plugins.

Reply:
We should be careful with that, not everything has to be idempotent. Idempotence is important for the deployment and impl store modules, and it can have interesting properties for the user, but it's not a goal in general.

That said, I think it's a good idea but we should only ignore files with `manifestVersion >= 3.0`.